plugin/helpers_kv.go

```diff
@@ -129,3 +130,91 @@ func (p *HelpersImpl) KVSetWithExpiryJSON(key string, value interface{}, expireI
 
 	return nil
 }
+
+type kvListOptions struct {
+	KVListPerPage int
+	Prefix        string
+	Checker       func(key string) (keep bool, err error)
+}
+
+// KVListOption represents a single input option for KVListWithOptions
+type KVListOption func(*kvListOptions)
+
+// WithPrefix will filter out any key that doesn't start with the given string.
```

Comment:
Nit pick: Maybe invert the wording and say 
```suggestion
// WithPrefix only return keys that start with the given string.
```